src/cmap/connection.ts

```diff
@@ -709,7 +709,14 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
       }
     }
 
-    if (this.socket.write(buffer)) return;
+    try {
+      if (this.socket.write(buffer)) return;
+    } catch (writeError) {
+      const cause = writeError as Error;
```

Comment:
Any reason for the cast?  technically you can throw non-Error stuff in Javascript but we usually just do the following:

```typescript
catch (error) {
    throw new SomeOtherError(<message>, { cause: error });
}
```